pyaerocom/test/test_colocation_auto.py

```diff
@@ -0,0 +1,181 @@
+    coldata = col_tm5_aero.data[col_tm5_aero.model_id][col_tm5_aero.obs_vars[0]]
+    assert isinstance(coldata, ColocatedData)
+
+def test__run_gridded_ungridded(init_obsdata_paths, col_tm5_aero):
+    obs_vars = col_tm5_aero.obs_vars[0]
+    colocated_dict = col_tm5_aero._run_gridded_ungridded()
+    assert isinstance(colocated_dict[obs_vars], ColocatedData)
+
+def test__run_gridded_gridded(init_obsdata_paths, col_tm5_aero):
+    obs_vars = col_tm5_aero.obs_vars[0]
+    col_tm5_aero.obs_id = col_tm5_aero.model_id
+    colocated = col_tm5_aero._run_gridded_gridded(obs_vars)
+    assert isinstance(colocated[obs_vars], ColocatedData)
+    assert isinstance(col_tm5_aero.data, dict)
+    assert col_tm5_aero.data == {}
+
+def test_colocator_filter_name():
+    with does_not_raise_exception():
+        col = Colocator(filter_name='WORLD')
+    with pytest.raises(Exception):
+        col = Colocator(filter_name='invalid')
+
+def test_colocator_basedir_coldata(tmpdir):
+    basedir = os.path.join(tmpdir, 'test')
+    col = Colocator(raise_exceptions=True)
+    col.basedir_coldata = basedir
+    assert not os.path.isdir(basedir)
+
+def test_colocator__find_var_matches(col):
+    r = ReadGridded('TM5-met2010_CTRL-TEST')
+    with pytest.raises(DataCoverageError):
+        col._find_var_matches('invalid', r)
+    var_matches = col._find_var_matches('od550aer', r)
+    assert var_matches == {'od550aer': 'od550aer'}
+
+    obs_vars = 'conco3'
+    col.obs_vars = [obs_vars]
+    col.model_use_vars = {obs_vars : 'od550aer'}
+    var_matches = col._find_var_matches('conco3', r)
+    assert var_matches == {'od550aer' : 'conco3'}
+
+def test_colocator_dir():
+    col = Colocator(raise_exceptions=True)
+    assert isinstance(dir(col), list)
+
+### Remove?
+
+def test_colocator_read_ungridded(): # Never called, delete?
+    col = Colocator(raise_exceptions=True)
+    obs_id = 'AeronetSunV3L2Subset.daily'
+    obs_var = 'od550aer'
+    col.obs_filters = {'longitude' : [-30, 30]}
+    col.obs_id = obs_id
+    col.read_opts_ungridded = {'last_file' : 10}
+    with pytest.raises(DataCoverageError):
+        data = col.read_ungridded(obs_var) # Why is this raised?
+        # read_ungridded expects a list of one or more variables!
+        # Should check for str and convert to list?
+        # Should read_ungridded and read_model_data have same arguments?
+    data = col.read_ungridded([obs_var])
+    assert isinstance(data, UngriddedData)
+
+    col.read_opts_ungridded = None
+    col.obs_vars = ['od550aer']
+    with does_not_raise_exception():
+        data = col.read_ungridded()
+    col.obs_vars = ['invalid']
+    with pytest.raises(DataCoverageError):
+        data = col.read_ungridded()
+
+def test_colocator_read_model_data(): # Never called, delete?
+    col = Colocator(raise_exceptions=True)
+    model_id = 'TM5-met2010_CTRL-TEST'
+    col.model_id = model_id
+    data = col.read_model_data('od550aer')
+    assert isinstance(data, GriddedData)
```

Comment:
Please keep, need this method in some of my scripts.

Reply:
I mean the actual method in `Colocator` of course, not the test method :) But having a test for this is great as well!